Adding ASSIN2 (portuguese) - STS and PC tasks

wissam-sib opened the pull request:
Adding Assin 2 RTE subtask to have a first portuguese dataset for Pair Classification. Additionally adding the STS part of the challenge.

- [ ] I have tested that the dataset runs with the `mteb` package.
- [ ] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [ ] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [ ] `intfloat/multilingual-e5-small`
- [ ] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [ ] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [ ] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [ ] Run tests locally to make sure nothing is broken using `make test`. 
- [ ] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

imenelydiaker commented on mteb/tasks/PairClassification/por/Assin2RTE.py:
```diff
@@ -0,0 +1,58 @@
+
+class Assin2RTE(AbsTaskPairClassification):
+    metadata = TaskMetadata(
+        name="assin2-rte",
```
Maybe name the task as the class to follow the same convention
```suggestion
        name="Assin2RTE",
```

imenelydiaker commented on mteb/tasks/PairClassification/por/Assin2RTE.py:
```diff
@@ -0,0 +1,58 @@
+        reference="https://link.springer.com/chapter/10.1007/978-3-030-41505-1_39",
+        type="PairClassification",
+        category="s2s",
+        eval_splits=["test"],
```
we can add the validation sest also ?
```suggestion
        eval_splits=["validation", "test"],
```

wissam-sib replied:
as the test set is already big enough (>2048), we can keep test only if that's fine with you

imenelydiaker replied:
okay great!

imenelydiaker commented on mteb/tasks/PairClassification/por/Assin2RTE.py:
```diff
@@ -0,0 +1,58 @@
+            year={2020},
+            organization={Springer}
+        }""",
+        n_samples={"test": 2448},
+        avg_character_length={"test": 53.55},
```
If you add the validation set, add info here also

imenelydiaker commented on mteb/tasks/STS/por/Assin2STS.py:
```diff
@@ -0,0 +1,52 @@
+
+class Assin2STS(AbsTaskSTS):
+    metadata = TaskMetadata(
+        name="assin2-sts",
```
Same as above, for consistency

imenelydiaker commented on mteb/tasks/STS/por/Assin2STS.py:
```diff
@@ -0,0 +1,52 @@
+        return metadata_dict
+
+    def dataset_transform(self) -> None:
+        self.dataset = self.dataset.rename_column("premise", "sentence1")
+        self.dataset = self.dataset.rename_column("hypothesis", "sentence2")
+        self.dataset = self.dataset.rename_column("relatedness_score", "score")
```
```suggestion
        self.dataset = self.dataset.rename_columns({"premise": "sentence1", "hypothesis": "sentence2", "relatedness_score": "score"})
```

wissam-sib replied:
good point

imenelydiaker reviewed (commented):
LGTM! Thanks for your contribution, I left some comments below and please add points 🙂

imenelydiaker commented:
Please check that there no taks in the multilingual folders handling this language before adding bonus points

wissam-sib commented:
@imenelydiaker Committed your suggestions, fixed the name of Assin2STS as well. Reran the evals just to make sure. Added the points. ready to merge.

> Please check that there no taks in the multilingual folders handling this language before adding bonus points

I checked the multilingual tasks and Portuguese is not there for Pair Classification only.

imenelydiaker reviewed (approved):
LGTM! Let's merge then
